Quick Sort MPI (Optimized with -O2)
Number of cores: 1
Quick Sort Complete!
Execution Time: 14388 milliseconds
std::sort Time: 11937 milliseconds
Pass the sorting result check!
Number of cores: 2
Quick Sort Complete!
Execution Time: 12875 milliseconds
std::sort Time: 15933 milliseconds
Pass the sorting result check!
Number of cores: 4
Quick Sort Complete!
Execution Time: 9124 milliseconds
std::sort Time: 15903 milliseconds
Pass the sorting result check!
Number of cores: 8
Quick Sort Complete!
Execution Time: 7890 milliseconds
std::sort Time: 15887 milliseconds
Pass the sorting result check!
Number of cores: 16
Quick Sort Complete!
Execution Time: 7905 milliseconds
std::sort Time: 15970 milliseconds
Pass the sorting result check!
Number of cores: 32
Quick Sort Complete!
Execution Time: 8367 milliseconds
std::sort Time: 16022 milliseconds
Pass the sorting result check!

Bucket Sort MPI (Optimized with -O2)
Number of cores: 1
Bucket Sort Complete!
Execution Time: 12773 milliseconds
std::sort Time: 11810 milliseconds
Pass the sorting result check!
Number of cores: 2
Bucket Sort Complete!
Execution Time: 8617 milliseconds
std::sort Time: 15147 milliseconds
Pass the sorting result check!
Number of cores: 4
Bucket Sort Complete!
Execution Time: 4440 milliseconds
std::sort Time: 15045 milliseconds
Pass the sorting result check!
Number of cores: 8
Bucket Sort Complete!
Execution Time: 2832 milliseconds
std::sort Time: 15060 milliseconds
Pass the sorting result check!
Number of cores: 16
Bucket Sort Complete!
Execution Time: 1827 milliseconds
std::sort Time: 15171 milliseconds
Pass the sorting result check!
Number of cores: 32
Bucket Sort Complete!
Execution Time: 1296 milliseconds
std::sort Time: 15205 milliseconds
Pass the sorting result check!

Odd-Even Sort MPI (Optimized with -O2)
Number of cores: 1
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

srun: error: node37: task 0: Aborted (core dumped)
Number of cores: 2
terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

srun: error: node37: tasks 0-1: Aborted (core dumped)
Number of cores: 4
terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
  what():  Invalid argument, should be: ./executable vector_size


srun: error: node37: tasks 0-3: Aborted (core dumped)
Number of cores: 8
terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
  what():  Invalid argument, should be: ./executable vector_size


terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

srun: error: node37: tasks 0,2,4,6: Aborted (core dumped)
srun: error: node37: tasks 1,3,5,7: Aborted
Number of cores: 16
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size


terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
  what():  Invalid argument, should be: ./executable vector_size


terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
  what():  Invalid argument, should be: ./executable vector_size


terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

srun: error: node37: tasks 0-3,5,7,10-15: Aborted
srun: error: node37: tasks 4,6,8-9: Aborted (core dumped)
Number of cores: 32
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size


terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument, should be: ./executable vector_size

srun: error: node37: tasks 0,2-8,10-11,13,15-31: Aborted
srun: error: node37: tasks 1,9,12,14: Aborted (core dumped)

